agent/consul/stream/event.go

```diff
@@ -0,0 +1,44 @@
+/*
+Package stream provides a publish/subscribe system for events produced by changes
+to the state store.
+*/
+package stream
+
+import "fmt"
+
+// Topic is an identifier that partitions events. A subscription will only receive
+// events which match the Topic.
+type Topic fmt.Stringer
```

Comment:
I went with `fmt.Stringer` for now, but I expect this interface could change when we add protobuf types. My rational for using an interface here instead of a concrete type is:

* nothing in the `stream` package cares about the type
* once we add the protobuf types we should be able to use the protobuf `Topic` type here, removing the need to translate the types and have them topics duplicated in two packages
* I still expect there to be a list of topics defined as "constants" (technically vars, but effectively constants), but they would be in either `state` or the protobuf package. It should be easy to see the full list from there, so I think we still get the same benefits.

I think this separation makes the `stream` package re-usable. It is a fairly generic pub/sub system that may be usable in other places.

Reply:
Another consideration is that an `EventPublisher` will only accept subscriptions for `Topic`s identified in the mapping of `SnapshotHandlers`. So the global list of possible topics isn't really important. Any given EventPublisher will only handle Topics defined that it is told about.